src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/integration/IntegrationTestBase.kt

```diff
@@ -149,6 +153,17 @@ abstract class IntegrationTestBase {
     }
   }
 
+  fun WebTestClient.ResponseSpec.errorResponse(httpStatus: HttpStatus = HttpStatus.BAD_REQUEST) =
+    expectStatus().isEqualTo(httpStatus)
+      .expectBody<ErrorResponse>()
+      .returnResult().responseBody!!
+
+  final inline fun <reified T> WebTestClient.ResponseSpec.successResponse(httpStatus: HttpStatus = HttpStatus.OK) =
+    expectStatus().isEqualTo(httpStatus)
+      .expectHeader().contentType(MediaType.APPLICATION_JSON)
+      .expectBody(T::class.java)
+      .returnResult().responseBody!!
```

Comment:
i like these helper functions!
i think HttpStatus.OK default for success response makes sense.
but HttpStatus.BAD_REQUEST default for error seems a bit random?

Reply:
I know what you mean, Bill. Bad request is the most common but there are lots of 4xx and 5xx tests. Adding `fun WebTestClient.ResponseSpec.badRequestResponse() = fun WebTestClient.ResponseSpec.errorResponse(httpStatus: HttpStatus = HttpStatus.BAD_REQUEST)` is one option but would encourage devs to add a function for each code. On balance I think you're right that removing the bad request default is best